src/NSubstitute.Analyzers.CSharp/DiagnosticAnalyzers/CallInfoAnalyzer.cs

```diff
@@ -86,5 +86,11 @@ protected override bool CanCast(Compilation compilation, ITypeSymbol sourceSymbo
         {
             return compilation.ClassifyConversion(sourceSymbol, destinationSymbol).Exists;
         }
+
+        protected override bool IsAssignableTo(Compilation compilation, ITypeSymbol fromSymbol, ITypeSymbol toSymbol)
+        {
+            var conversion = compilation.ClassifyConversion(fromSymbol, toSymbol);
+            return conversion.Exists && conversion.IsExplicit == false && conversion.IsNumeric == false;
```

Comment:
Can you please explain why `conversion.IsNumeric` is excluded? (I don't know anything about `Conversion` so sorry if this is a silly question.)

As per other comment, is there a reason for preferring `x == false` compared with `!x`? (I don't have a strong view on this, just curious)

Reply:
There is implicit conversion between some numeric types. For instance this is valid c#
````c#
int x = 1;
decimal xx = x;
````
so conversion is numeric and exists. However it doesnt mean that type is assignable to other type in terms of NSubstitute check - as NSubstitute checks this as `type.IsAssignableFrom(otherType)`

Here is a simple repro
````c#
using NSubstitute;
using System.Collections.Generic;
using NSubstitute.ReturnsExtensions;
using Xunit;

namespace MyNamespace
{
    public interface IFoo
    {
        int Bar(out int x);
    }

    public class FooTests
    {
        [Fact]
        public void Test()
        {
            var substitute = Substitute.For<IFoo>();

            int i;
            substitute.Bar(out i).Returns(callInfo =>
            {
                callInfo.ArgAt<decimal>(0);
                return 1;
            });

            substitute.Bar(out var x);
        }
    }
}
````
![rider64_2019-01-02_21-54-24](https://user-images.githubusercontent.com/7378346/50612078-0e2c7f00-0ed9-11e9-8adc-48c46d3d656f.png)